src/ansi-c/expr2c.cpp

```diff
@@ -860,6 +840,42 @@ std::string expr2ct::convert_array_type(
     src.subtype(), qualifiers, declarator_str+array_suffix);
 }
 
+std::string expr2ct::convert_pointer_type(
+  const typet &src,
+  const c_qualifierst &qualifiers,
+  const std::string &declarator)
+{
+    c_qualifierst new_qualifiers(qualifiers);
```

Comment:
No, it is. If you see above, in the code block removed from if statement at line `491`, the code in the consequent statement is defining its own local `c_qualifierst` variable in order to call the method `read` on, as the relevant variable is still passed to the method that encloses that statement with a `const` qualifier, not allowing you to call `read` on it. If I don't do this, I get errors like 
```
/home/fotis/Devel/cbmc-public/src/ansi-c/expr2c.cpp:849: error: member function 'read' not viable: 'this' argument has type 'const c_qualifierst', but function is not marked const
    new_qualifiers.read(src);
    ^~~~~~~~~~~~~~
```

In other words it's required because we can't call the method on a const object that is referenced in the enclosing environment at both instances.